src/pages/Vote/Governance/index.tsx

```diff
@@ -1,19 +1,95 @@
 /** @jsxImportSource @emotion/react */
-import React from 'react';
+import React, { useState } from 'react';
 import { Typography } from '@mui/material';
 import { useTranslation } from 'translation';
+import { useGetProposals } from 'clients/api';
+import { GovernanceProposal, Icon, Spinner, TextButton, Tooltip, Pagination } from 'components';
+import { IProposal } from 'types';
 import { useStyles } from './styles';
 
-export const GovernanceUi: React.FC = () => {
+interface IGovernanceUiProps {
+  proposals: IProposal[];
+  isLoading: boolean;
+  total: number | undefined;
+  limit: number;
+  setCurrentPage: (page: number) => void;
+}
+
+export const GovernanceUi: React.FC<IGovernanceUiProps> = ({
+  proposals,
+  isLoading,
+  total,
+  limit,
+  setCurrentPage,
+}) => {
   const { t } = useTranslation();
   const styles = useStyles();
   return (
     <div css={styles.root}>
-      <Typography variant="h4">{t('vote.governanceProposals')}</Typography>
+      <div css={[styles.header, styles.bottomSpace]}>
+        <Typography variant="h4">{t('vote.governanceProposals')}</Typography>
+        <div css={styles.createProposal}>
+          <TextButton css={styles.marginless}>{t('vote.createProposal')}</TextButton>
+          <Tooltip title={t('vote.requiredVotingPower')} css={styles.infoIcon}>
+            <Icon size="16px" name="info" />
+          </Tooltip>
+        </div>
+      </div>
+      {isLoading && <Spinner />}
+      <div>
+        {proposals.map(
+          ({
+            id,
+            description,
+            state,
+            endDate,
+            forVotesWei,
+            abstainedVotesWei,
+            againstVotesWei,
+          }) => (
+            <GovernanceProposal
+              key={id}
+              css={styles.bottomSpace}
+              proposalNumber={id}
+              proposalDescription={description}
+              proposalState={state}
+              endDate={endDate}
+              forVotesWei={forVotesWei}
+              againstVotesWei={againstVotesWei}
+              abstainedVotesWei={abstainedVotesWei}
+            />
+          ),
+        )}
+      </div>
+      {total && (
```

Comment:
Do you think you could place a loader somewhere on the page when a new page is being loaded?
I described this UX on your PR for the History screen: I clicked on a new page button in the pagination component, the page scrolled up and for 1-2 seconds nothing happened. The old page data was still showing and I didn't have a visual indication something was happening

Reply:
yeah, I'll take care of that on the other PR